buildSrc/src/main/groovy/com.google.api-ads.java-conventions.gradle

```diff
@@ -58,67 +63,165 @@ dependencies {
-          '\033[0;35m--example="basicoperations.GetCampaigns"\033[0m\n\n' +
-          e.message)
+        'you provide a valid example identifier? E.g. ' +
+        '\033[0;35m--example="basicoperations.GetCampaigns"\033[0m\n\n' +
+        e.message)
     }
   }
 
-  public String getExampleArguments() {
-    return exampleArguments
-  }
-
   private void logQuietMessage(String message) {
     logger.quiet(message)
   }
+}
+
+class ListExamplesTask extends DefaultTask {
+
+  // The base directory containing all examples.
+  private final String examplesDirectory =
+    'google-ads-examples/src/main/java/com/google/ads/googleads/examples/'
+
+  // Regex for extracting descriptions.
+  @Input
+  private final String regex =
+    '(?s)(/\\*\\*(?:(?!/\\*\\*).)*?\\*/)(?:(?:(?!\\*/).)*?)(?=public class )'
+
+  // Regex for comment markers.
+  @Input
+  private final String commentMarkers = '(/\\*\\*\\s+|\\*\\s|\\*/)'
+
+  // A list of subdirectories to exclude from the printout.
+  @Input
+  List<String> directoriesToExclude = ['utils']
+
+  // An optional subdirectory; if non-null, only examples in the specified
+  // subdirectory will be printed.
+  @Optional
+  @Input
+  private String subdirectory = null
+
+  ListExamplesTask() {
+    description = 'List all available code examples with package name ' +
+      'and description.'
+  }
 
+  String getSubdirectory() {
+    return this.subdirectory
+  }
+
+  String getRegex() {
+    return this.regex
+  }
+
+  String getCommentMarkers() {
+    return this.commentMarkers
+  }
+
+  List<String> getDirectoriesToExclude() {
+    return this.directoriesToExclude
+  }
+
+  @Option(option = 'subdirectory', description = 'Sets a subdirectory of ' +
+    'google-ads-examples, e.g. "basicoperations". Only examples in this ' +
+    'subdirectory will be printed. If not specified, all the examples ' +
+    ' will be printed.')
+  void setSubdirectory(String subdirectory) {
+    this.subdirectory = subdirectory.trim()
+  }
+
+  @TaskAction
+  void exec() {
+    String dir = subdirectory != null ?
+      examplesDirectory + subdirectory + '/' : examplesDirectory
+    int count = 0
+    int baseLength = examplesDirectory.length()
+
+    new File(dir).traverse(
+        type                : FileType.FILES,
+        preDir              : { if (it.name in directoriesToExclude)
+                                return FileVisitResult.SKIP_SUBTREE },
+        excludeNameFilter   : { it in directoriesToExclude },
+        nameFilter          : { it.endsWith('.java') },
+        sort                : { a, b -> a.name <=> b.name }
+    ) {
+        // Strip out the base directory, truncate the .java file extension,
+        // and replace the / with . to indicate package name.
+        String className = it.path.substring(baseLength,
+          it.path.length()-5).replaceAll('/','.')
+
+        // Fetch the example description.
+        Matcher matcher = it.text =~ getRegex()
```

Comment:
Extracting the description looks like it could be quite brittle. I wonder if it would be easier to maintain if we just printed the class name?

Reply:
Certainly true regarding the maintainability, but IMO printing the description gives useful context. As is, all examples do use this format consistently.